### PR TITLE
Refactor dashboard features to use `selector-observer`

### LINKED ISSUE #3469: `open-issue-to-latest-comment` doesn't work consistently
Make the “comment” icon and “star”icon in `https://github.com/` clickable.
1. Makes the “comment” icon in `https://github.com/` link to the latest comment of the issue.
image
2. Makes the “comment” icon in `https://github.com/` link to the Stargazers page of the repository.
image

### PR DESCRIPTION
**Pull request by yakov116**
1. LINKED ISSUES:
   Fixes #3469 

2. TEST URLS:
   https://github.com/

3. SCREENSHOT:
   There should be no changes

### CODE REVIEW

File: source/features/linkify-labels-on-dashboard.tsx [hunk 1]
```diff
@@ -8,9 +8,13 @@ import {wrap} from '../helpers/dom-utils';
...
 function init(): void {
-	observe('.js-recent-activity-container :not(a) > .IssueLabel', {
+	const labelClass = [
+		'.js-recent-activity-container :not(a) > .IssueLabel', // Recent activity
+		'.js-all-activity-header + div :not(a) > .IssueLabel' // Newsfeed
+	].join();
+	observe(labelClass, {
 		add(label) {
-			const activity = label.closest('li')!;
+			const activity = label.closest('li')! ?? label.parentElement; // Newsfeed
```
Comment (yakov116) [inline source/features/linkify-labels-on-dashboard.tsx]: Does this have to go in its own PR?

**Comment by fregante** [comment]
Can you also remove “default listeners” the line from features/index? 

Are there any other features with isDashboard?

**Comment by yakov116** [comment]
Only infinite-scroll but that is too complex for me to try and change,

File: source/features/open-issue-to-latest-comment.tsx [hunk 2]
```diff
@@ -12,13 +13,15 @@ function init(): void {
...
-		icon.parentElement!.append(link);
-		link.append(icon, icon.nextSibling!);
-	}
+	observe('.js-recent-activity-container :not(a) > div > .octicon-comment', {
+		add(icon) {
+			const url = icon.closest('li')!.querySelector('a')!.pathname + '#partial-timeline';
+			const link = <a className="muted-link" href={url}/>;
+			icon.parentElement!.classList.remove('col-1'); // Also fix extra space added by GitHub #3174
+			icon.parentElement!.append(link);
+			link.append(icon, icon.nextSibling!);
```
Comment (fregante) [inline source/features/open-issue-to-latest-comment.tsx:22]: I think this could be also replaced with
```suggestion
			wrapAll(link [icon, icon.nextSibling!);
```

File: source/features/linkify-labels-on-dashboard.tsx (hunk 1 above)
Comment (fregante) [reply source/features/linkify-labels-on-dashboard.tsx]: Yes. It's not correct now because `isPR` will never match.

**Comment by fregante** [comment]
> Only infinite-scroll but that is too complex for me to try and change,

That's fine as is, it only need the explicit `onNewsfeedLoad` now after https://github.com/sindresorhus/refined-github/pull/3521/commits/00f78fe56fa4f249c5a0b5141bf8fd5cb023ee46

File: source/features/linkify-labels-on-dashboard.tsx (hunk 1 above)
Comment (yakov116) [reply source/features/linkify-labels-on-dashboard.tsx]: What?

Comment (fregante) [reply source/features/linkify-labels-on-dashboard.tsx]:
```js
const isPR = select.exists('.octicon-git-pull-request', activity);
```

There's never a `octicon-git-pull-request` inside `label.parentElement`. Just drop this change from this PR for now.

**Comment by fregante** [comment]
Likewise, the same can be done for any [every/most `hasComments` feature](https://github.com/sindresorhus/refined-github/search?q=hasComments&type=Code) in a future PR. (initially mentioned in https://github.com/sindresorhus/refined-github/pull/3407#pullrequestreview-478157181)

File: source/features/linkify-labels-on-dashboard.tsx (hunk 1 above)
Comment (yakov116) [reply source/features/linkify-labels-on-dashboard.tsx]: I will I forgot to mention it only is there if `hide-usless-feeds` is disabled. Regardless I am reverting.

Comment (fregante) [reply source/features/linkify-labels-on-dashboard.tsx]: Indeed. Let's not add that at all

**Comment by yakov116** [comment]
@fregante I had an emergency come up I may be off for a day or 2. I will try to finish it soon.

**Comment by yakov116** [comment]
Done

File: source/features/open-issue-to-latest-comment.tsx [hunk 3]
```diff
@@ -12,13 +14,14 @@ function init(): void {
...
-		const url = icon.closest('li')!.querySelector('a')!.pathname + '#partial-timeline';
-		const link = <a className="muted-link" href={url}/>;
-		icon.parentElement!.classList.remove('col-1'); // Also fix extra space added by GitHub #3174
-		icon.parentElement!.append(link);
-		link.append(icon, icon.nextSibling!);
-	}
+	observe('.js-recent-activity-container :not(a) > div > .octicon-comment', {
+		add(icon) {
+			const url = icon.closest('li')!.querySelector('a')!.pathname + '#partial-timeline'
+
```
Comment (yakov116) [inline source/features/open-issue-to-latest-comment.tsx:20]:
```suggestion
```

File: source/features/open-issue-to-latest-comment.tsx [hunk 4]
```diff
@@ -12,13 +14,13 @@ function init(): void {
...
-	for (const icon of select.all('.js-recent-activity-container :not(a) > div > .octicon-comment')) {
-		const url = icon.closest('li')!.querySelector('a')!.pathname + '#partial-timeline';
-		const link = <a className="muted-link" href={url}/>;
-		icon.parentElement!.classList.remove('col-1'); // Also fix extra space added by GitHub #3174
-		icon.parentElement!.append(link);
-		link.append(icon, icon.nextSibling!);
-	}
+	observe('.js-recent-activity-container :not(a) > div > .octicon-comment', {
+		add(icon) {
+			const url = icon.closest('li')!.querySelector('a')!.pathname + '#partial-timeline'
```
Comment (yakov116) [inline source/features/open-issue-to-latest-comment.tsx:19]:
```suggestion
			const url = icon.closest('li')!.querySelector('a')!.pathname + '#partial-timeline';
```

File: source/features/infinite-scroll.tsx [hunk 5]
```diff
@@ -65,6 +66,8 @@ void features.add({
 	include: [
 		pageDetect.isDashboard
 	],
-	onlyAdditionalListeners: true,
```
Comment (fregante) [inline source/features/infinite-scroll.tsx:68]: I think this still applies. It **only** needs onNewsfeedLoad

Comment (yakov116) [reply source/features/infinite-scroll.tsx:68]: I added it

Comment (fregante) [reply source/features/infinite-scroll.tsx:68]: Otherwise, this feature could be rewritten as well. Drop `findButton` everywhere, and add something like this in `init`:

```js
observe('.ajax-pagination-btn', {
  add(button) {
    inView.observe(button);
  }
});
```

Comment (yakov116) [reply source/features/infinite-scroll.tsx:68]: I am too scared to touch it its waay beyond my understanding.

Comment (yakov116) [reply source/features/infinite-scroll.tsx:68]: I did it I have no idea if it works since I dont have enough events 😃

Comment (yakov116) [reply source/features/infinite-scroll.tsx:68]: Tested on isProfileMainTab, it works

**Comment by yakov116** [comment]
What about `show-names`?

**Comment by fregante** [comment]
This doesn't appear to be affected by https://github.com/josh/selector-observer/issues/30 👍

**Comment by yakov116** [comment]
@fregante I restored the listener for now. I will come back to it.

Are we good to merge now?

**Review by fregante — APPROVED** [review]
